### PR TITLE
Feature/transactions

### PR DESCRIPTION
**Pull request by markelrep**

### CODE REVIEW

File: src/api/handlers/transactions.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+// HandlePatch is setting or changing a category for specify transactionResponse
+func (handler *TransactionsHandler) HandlePatch(c *gin.Context) {
+	var req patchCategoryRequest
+	err := c.BindJSON(&req)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, errorResponse{Message: "Bad request", Error: err.Error()})
+		return
+	}
+	transactionID := c.Param("transactionID")
```
Comment (SuddenGunter) [inline src/api/handlers/transactions.go]: if it's not there?
or if it's invalid objectId?
like PATCH `/transacions/blablabla` ?

File: src/transactions/repository_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/stretchr/testify/assert"
+
+	"github.com/lungria/mono"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+var txn1 = Transaction{
+	ID:        primitive.NewObjectID(),
+	AccountID: "fCfTvZgMihvz_URzZSqVEf",
+	StatementItem: mono.StatementItem{
```
Comment (SuddenGunter) [inline src/transactions/repository_test.go]: I don't think this is the best approach.
https://www.mongodb.com/blog/post/6-rules-of-thumb-for-mongodb-schema-design-part-1
mongoDb docs suggest to store data in the same way you use it.
I suggest that basic structure of the model of all our transactions must be *exactly* the same as we use on `GET /transactions` endpoint:
```
	ID   
	CategoryId // or maybe embed full category object? not sure
	AccountId
        DateTime // In UTC, based on when transaction really occured - https://stackoverflow.com/a/15952652/6878401 - ISO 8601 for JSON and mongodb Date type https://docs.mongodb.com/manual/reference/bson-types/#document-bson-type-date 
        Amount // In UAH. I suggest to postpone multi-currency support. we still can store this info in BankTransaction
        Description
```
We may, if we need to store monobank transaction as an embedded field, as it done here.
I suggest  rename it to BankTransaction or something like that - there is 85% chance that this app would also support PrivatBank transactions. They  don't support webhooks at the moment, but we can export CSV and parse it

File: src/webhooks/repository.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"go.uber.org/zap"
+
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+const (
+	transactionsCollection = "transactions"
+)
+
+// Repository define all methods which do some work with database
```
Comment (SuddenGunter) [inline src/webhooks/repository.go]: >all methods which do some work with database

I think this is to broad and generic

File: src/webhooks/repository.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		return nil, errors.New("logger must not be nil")
+	}
+	return &WebHookRepository{
+		collection: db.Collection(transactionsCollection),
+		logger:     logger,
+	}, nil
+}
+
+// SaveOneHook save one Transaction to MongoDB
+func (repo *WebHookRepository) SaveOneHook(webhook *WebHook) error {
```
Comment (SuddenGunter) [inline src/webhooks/repository.go]: Insert?

File: src/api/handlers/categories.go [hunk 5]
```diff
@@ -14,23 +14,25 @@ import (
...
 
-type GetAllCategoriesResponse struct {
+type getAllCategoriesResponse struct {
 	Categories []categories.Category `json:"categories"`
 }
 
-type InsertCategoryResponse struct {
-	Id primitive.ObjectID `json:"id"`
+type insertCategoryResponse struct {
+	ID primitive.ObjectID `json:"id"`
```
Comment (SuddenGunter) [inline src/api/handlers/categories.go]: why rename?

Comment (markelrep) [reply src/api/handlers/categories.go]: This type use only by the handler, therefore it shouldn't be a public

File: src/api/handlers/transactions.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+func (handler *TransactionsHandler) HandlePatch(c *gin.Context) {
+	var req patchCategoryRequest
+	err := c.BindJSON(&req)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, errorResponse{Message: "Bad request", Error: err.Error()})
+		return
+	}
+
+	transactionID := c.Param("transactionID")
+	tObjID, err := primitive.ObjectIDFromHex(transactionID)
```
Comment (SuddenGunter) [inline src/api/handlers/transactions.go]: what if transactionId == nil?

File: src/transactions/repository_test.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		CurrencyCode:    826,
+		CommissionRate:  0,
+		CashbackAmount:  0,
+		Balance:         90,
+	},
+}
+
+var txn2 = Transaction{
+	ID:        primitive.NewObjectID(),
+	Category:  "Shopping",
```
Comment (SuddenGunter) [inline src/transactions/repository_test.go]: use categoryId

File: src/transactions/transaction.go [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package transactions
+
+import (
+	"github.com/lungria/mono"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Transaction represents a model of transactions in database
+type Transaction struct {
```
Comment (SuddenGunter) [inline src/transactions/transaction.go]: amount?
DateTime?

Comment (SuddenGunter) [reply src/transactions/transaction.go]: Description?

File: src/api/handlers/transactions.go (hunk 6 above)
Comment (markelrep) [reply src/api/handlers/transactions.go]: gin returns "404 Not found" in this case

**Pull request merged**